fastparquet/writer.py

```diff
@@ -570,6 +571,9 @@ def write_column(f, data, selement, compression=None):
             total_uncompressed_size=uncompressed_size,
             total_compressed_size=compressed_size)
     if cats:
+        # all chunks describe if they are categorical
+        cmd.key_value_metadata = [
+            parquet_thrift.KeyValue(key="fastparquet.cats", value=str(ncats))]
```

Comment:
Do you have any interest in figuring out a solution that is not restricted to only fastparquet?  Other projects might also benefit from this sort of metadata.